components/patient-data/impl/src/main/java/org/phenotips/data/internal/PhenoTipsPatient.java

```diff
@@ -325,7 +333,8 @@ public JSONObject toJSON(Collection<String> onlyFieldNames)
             result.element(JSON_KEY_DISORDERS, diseasesToJSON());
         }
 
-        for (PatientDataController<?> serializer : this.serializers) {
+
+        for (PatientDataController<?> serializer: this.serializers.values()) {
```

Comment:
As long as nothing depends on the ordering in which the serializer JSONs are written being consistent between calls, this is okay. If there's any possibility that consistent ordering might be a good thing (e.g., if this data is ever diff'd), then I'd suggest replacing the `HashMap` with a `TreeMap`.